examples/SpringHibernateBalancedReaderOneDataSourceExample/CUSTOM_ANNOTATIONS.md

```diff
@@ -0,0 +1,264 @@
+# Custom Annotations Documentation
+
+## Overview
+
+The SpringHibernateBalancedReaderOneDataSourceExample demonstrates a **single datasource approach** for load-balanced read-write and read-only connections using the AWS Advanced JDBC Driver's built-in Read/Write Splitting Plugin. Unlike the TwoDataSourceExample, this approach does **not** use custom annotations for datasource routing.
+
+## Key Difference from TwoDataSourceExample
+
+| Aspect | OneDataSourceExample | TwoDataSourceExample |
+|--------|---------------------|----------------------|
+| **Datasources** | Single datasource | Two separate datasources (writer + reader) |
+| **Routing Mechanism** | AWS JDBC Driver's Read/Write Splitting Plugin | Custom `@WithLoadBalancedReaderDataSource` annotation |
+| **Configuration** | Connection string with `wrapperProfileName=F0` | Spring's `AbstractRoutingDataSource` + AOP |
+| **Custom Annotations** | None (uses standard Spring annotations) | Custom annotation with AspectJ interception |
+| **Complexity** | Simpler, driver-managed | More complex, application-managed |
+
+## Standard Annotations Used
+
+### @Transactional(readOnly = true)
+
+The key to routing in this example is Spring's standard `@Transactional` annotation with the `readOnly` parameter:
+
+```java
+@Retryable(value = { ShouldRetryTransactionException.class }, maxAttempts = 3)
```

Comment:
Probably need to replace `ShouldRetryTransactionException` with `FailoverSQLException`